<?xml version="1.0" encoding="UTF-8"?><?oxygen RNGSchema="http://www.oasis-open.org/docbook/xml/5.0/rng/docbookxi.rng" type="xml"?>
<ns6514:section xmlns:ns6514="http://docbook.org/ns/docbook" xml:id="resources.information.technologies.java.j2se.exceptionHandling.uncaughtExceptionHandlers.settingDefaultUncaughtExceptionHandler.settingDefaultUncaughtExceptionHandler">  
  <ns6515:title xmlns:ns6515="http://docbook.org/ns/docbook">Setting uncaught exception handler for thread</ns6515:title>
  <ns6516:para xmlns:ns6516="http://docbook.org/ns/docbook">
    To specify a default uncaught exception handler which will be used for all threads which have
    not got their own uncaught exception handlers,, we simply call the
    <ns6517:literal xmlns:ns6517="http://docbook.org/ns/docbook">setDefaultUncaughtExceptionHandler</ns6517:literal> on that thread class itself (a
    class/static service).
  </ns6516:para>
  <ns6518:para xmlns:ns6518="http://docbook.org/ns/docbook">
    This is illustrates in the following program
    <ns6519:programlisting xmlns:ns6519="http://docbook.org/ns/docbook">
<xi:include xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" parse="text" href="../../src/UncaughtExceptionTest2.java"/>      
    </ns6519:programlisting>
    whose output is something like this:
    <ns6520:programlisting xmlns:ns6520="http://docbook.org/ns/docbook">
MySpecialThread2 is starting ...
MySpecialThread3 is starting ...
MySpecialThread1 is starting ...
Yawn ...
Yawn ...
Yawn ...
Yawn ...
Yawn ...
Yawn ...
Aug 12, 2005 12:49:11 PM za.co.solms.exceptions.UncaughtExceptionLogger 
      uncaughtException
SEVERE: [Default] java.lang.RuntimeException: ^%!!??!^%%^??!## 
      thrown in thread MySpecialThread2 
      from file UncaughtExceptionTest2.java on line 19
Aug 12, 2005 12:49:11 PM za.co.solms.exceptions.UncaughtExceptionLogger 
      uncaughtException
SEVERE: [Default] java.lang.RuntimeException: ^%!!??!^%%^??!## 
      thrown in thread MySpecialThread3 
      from file UncaughtExceptionTest2.java on line 19
Aug 12, 2005 12:49:11 PM za.co.solms.exceptions.UncaughtExceptionLogger 
      uncaughtException
SEVERE:  java.lang.RuntimeException: ^%!!??!^%%^??!## 
      thrown in thread MySpecialThread1 
      from file UncaughtExceptionTest2.java on line 19
Yawn ...
Yawn ...
Yawn ...
Yawn ...
      </ns6520:programlisting>
  </ns6518:para>
</ns6514:section>